ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -634,6 +679,22 @@ let ocaml_on_request :
   | Client_request.TextDocumentOnTypeFormatting _ ->
     Fiber.return @@ Ok (None, state)
   | Client_request.SelectionRange req -> selection_range state req
+  | Client_request.UnknownRequest
+      { meth = "ocamllsp/didSwitchImplIntf"; params } ->
+    (* return a list of a single filepath to switch to if it exists or a list of
+       potential filepaths to create a file and to switch to *)
+    ( match params with
+    | Some (`List [ `String filepath; _ ]) ->
```

Comment:
We do this quite ugly and brittle pattern matching because we use `sendRequest` fn on the vscode extension side, which has the signature given below and takes a variable number of arguments (params to send). 

```typescript
(method) MessageConnection.sendRequest<unknown>(method: string, ...params: any[]): Promise<unknown> (+11 overloads)
```

So when the extension sends a single string param, the params sent have the form `["<string>", null]`. It would be great to get just a string on the ocaml-lsp side but I'm not sure how. 

cc @mnxn

Reply:
I took a look at the generated JS, and it seems that it is passing the optional and unit parameters as undefined to `sendRequest`:
```js
sendRequest(client, "ocamllsp/didSwitchImplIntf", $$document.fileName, undefined, undefined)
```

I pushed a fix that handles the parameters correctly and has the following generated code:
```js
function sendRequest(t, meth, data, token, param) {
  if (token !== undefined) {
    return t.sendRequest(meth, data, token);
  } else {
    return t.sendRequest(meth, data);
  }
}
```